gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/rest/HttpJsonServiceStubClassComposer.java

```diff
@@ -237,6 +237,18 @@ protected Expr createTransportSettingsInitExpr(
             .setArguments(Arrays.asList(TYPE_REGISTRY_VAR_EXPR))
             .build();
 
+    // This generates the Request Params Extractors twice (once in the APIMethodDescriptor
+    // and once in the CallSettings). This is to ensure backwards compatibility and allow
+    // for implicit routing headers with just HttpBindings.
+    if (method.shouldSetParamsExtractor()) {
```

Comment:
I think this should be already handled [here](https://github.com/googleapis/sdk-platform-java/blob/cc5724f04722e3d60eb9b4e02ecc1bdf5dd7a7fa/gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/grpc/GrpcServiceStubClassComposer.java#L273-L281)?

Reply:
Yeah, I believe that is for grpc. From the proposal, I think backwards compatibility (google.api.http) applies for both transports, so I've added it to the HttpJsonServiceStubClassComposer class as well.

Reply:
I think it was [migrated](https://github.com/googleapis/sdk-platform-java/blob/e7b2dbf24c4ecad5948340bc6078818167c059d5/gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/common/AbstractTransportServiceStubClassComposer.java#L1198-L1204) to the abstract composer as part of this PR? Any case that it couldn't handle?

Reply:
I believe it's still an abstract method inside the Abstract class: https://github.com/googleapis/sdk-platform-java/blob/7b6c9522fa0c16c5f98507bae9afff9a05bf7882/gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/common/AbstractTransportServiceStubClassComposer.java#L273

HttpJson also adds the `setParamsExtractor` only if `method.shouldSetParamsExtractor()`

Reply:
> I know where my confusion was coming from now, I was expecting this logic to be moved to the parent abstract class as well. I checked the logic in grpc composer and they look identical, can we move this common logic to the parent class? 

I believe there is a small different with the `TypeRegistry`, but I'll see if I can work around it.

> Also what does the comment mean `This generates the Request Params Extractors twice (once in the APIMethodDescriptor and and once in the CallSettings)`, I thought we only set it in the `CallSettings`? Is there a golden file with changes in `APIMethodDescriptor`?

I believe this was part of the original implementation and I forgot to remove.